Test_Challenge/Test_challenge.py

```diff
@@ -0,0 +1,87 @@
+import pygame
+
+class SyntaxChallenge:
+    def __init__(self):
+        self.id = "syntax_challenge"
+        pygame.init()
+        self.screen_width = 800
+        self.screen_height = 600
+        self.WHITE = (255, 255, 255)
+        self.BLACK = (0, 0, 0)
+        self.GRAY = (200, 200, 200)
+        self.screen = self.initialize_screen()
+        self.font = pygame.font.Font(None, 20)
+        self.instruction_text = self.get_challenge_instructions()
+        self.editor_rect = pygame.Rect(self.screen_width // 2, 0, self.screen_width // 2, self.screen_height)
+        self.running = True
+        self.users_code = ""
+        self.run_button = self.render_run_button()
+
+    def initialize_screen(self):
+        screen = pygame.display.set_mode((self.screen_width, self.screen_height))
+        pygame.display.set_caption('Syntax Challenge')
+        return screen
+
+    def get_challenge_instructions(self):
+        instruction_text = [
+            "Challenge Instructions:",
+            "Identify the correct syntax for a for-loop in Python.",
+            "Write your answer below and click RUN to submit."
+        ]
+        return instruction_text
+
+    def render_text_to_surface(self, surface, text, position, font, color):
+        lines = text.split('\n')
+        y_offset = 0
+        for line in lines:
+            text_surface = font.render(line, True, color)
+            surface.blit(text_surface, (position[0], position[1] + y_offset))
+            y_offset += text_surface.get_height() + 5
+
+    def validate_user_code(self):
+        # Here, you can define correct answers or use a simpler validation
+        correct_answers = ["for i in range(x):", "for element in iterable:"]
+        return self.users_code.strip() in correct_answers
+
+    def is_run_code_button_pressed(self, run_button):
+        mouse_pos = pygame.mouse.get_pos()
+        mouse_pressed = pygame.mouse.get_pressed()
+        if mouse_pressed[0] and run_button.collidepoint(mouse_pos):
+            return True
+        return False
+
+    def render_run_button(self):
+        button = pygame.Rect(30, self.screen_height - 60, 75, 30)
+        pygame.draw.rect(self.screen, self.BLACK, button)
+        self.render_text_to_surface(self.screen, "RUN", [button.x + 25, button.y + 5], self.font, self.WHITE)
+        return button
+
```

Comment:
add this public method to allow the main.py to check if this challamge has been completed

```
def is_completed(self):
       return self.completed
       
```